### ISSUE TITLE
ICN001 and N817 don't play with each other

### ISSUE THREAD

**Issue by ember91** [astral-sh/ruff#12916]
## List of keywords searched
 - ICN001
 - N817

## Minimal code snippets

`test1.py`

```python
import xml.etree.ElementTree as Et
```

`test2.py`

```python
import xml.etree.ElementTree as ET
```

## Invoked commands

```bash
ruff check test1.py test2.py
```

## Current ruff settings

```toml
select = ["ICN001", "N817"]
```

## Current ruff version

`ruff 0.6.0`

## Description

It seems like `N817` wants me to import it as `Et` while `ICN001` wants me to import it as `ET`. Perhaps it is ok that rules conflict in ruff. If so I guess this is not a bug.

**Comment by cclauss** [comment]
I also prefer 1.

% `echo "from xml.etree import ElementTree" | ruff check --select=ALL --ignore=D100,F401 -`
```
-:1:23: ICN001 `xml.etree.ElementTree` should be imported as `ET`
  |
1 | from xml.etree import ElementTree
  |                       ^^^^^^^^^^^ ICN001
```
% `echo "import xml.etree.ElementTree as ET" | ruff check --select=ALL --ignore=D100,F401 -`
```
-:1:8: N817 CamelCase `ElementTree` imported as acronym `ET`
  |
1 | import xml.etree.ElementTree as ET
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ N817
```

**Comment by AlexWaygood** [comment]
(1) sounds good to me too

**Comment by ember91** [comment]
Wow! You are so responsive in this repo. Hats off!

**Comment by hofbi** [comment]
@MichaReiser I still see `N817 CamelCase ElementTree imported as acronym ET` on `v0.6.1`. If I understand https://github.com/astral-sh/ruff/pull/12922 correctly, it should now be `import ElementTree as XET`? However, the autofix of `ICN001` still autofixes to `import ElementTree as ET`. Do I miss anything?

**Comment by cclauss** [comment]
I have never seen `XET` in the wild and the docs/tutorial recommends `ET`.

https://docs.python.org/3/library/xml.etree.elementtree.html#tutorial

**Comment by AlexWaygood** [comment]
@hofbi / @cclauss: We are definitely not recommending `XET` in this rule! 😄

The tests added in #12922 are specifically testing that the "conventional aliases" setting can be used to configure this rule. One fixture is being run with the "conventional aliases" setting being set to `"xml.etree.ElementTree": "XET"`, and the snapshot you see as a result shows you what would happen if you ran Ruff with that (strange) configuration

**Comment by MichaReiser** [comment]
> Ok makes sense for XET. Then do I have to configure anything so that both check play nicely together. Right now I still see the N817 error after the autofix adding ET.

There's a separate fix for `from ... import` that has not been released yet https://github.com/astral-sh/ruff/pull/12946 

If the incompatibility that you're seeing isn't related to `from ... imort`, please open a new issue and share a code example with us. Happy to take a look.

**Comment by hofbi** [comment]
Oh I missed that one. My issue is definitely https://github.com/astral-sh/ruff/issues/12940. So just waiting for the next release to get this fixed, thanks.

**Comment by MichaReiser** [comment]
No worries. I missed that one too when implanting the fix 🥲
